app/actions/AuthorisedAction.scala

```diff
@@ -33,68 +32,74 @@ import uk.gov.hmrc.http.{HeaderCarrier, SessionKeys}
+          errorHandler.internalServerError()(request)
+      }
     }
   }
 
-  private[actions] def individualAuthentication[A](block: AuthorisationRequest[A] => Future[Result], affinityGroup: AffinityGroup)
-                                                  (implicit request: Request[A], hc: HeaderCarrier): Future[Result] = {
+  private[actions] def individualAuthentication[A](sessionId: String,
+                                                   affinityGroup: AffinityGroup
+                                                  )(block: AuthorisationRequest[A] => Future[Result]
+                                                  )(implicit request: Request[A], hc: HeaderCarrier): Future[Result] = {
     authService.authorised().retrieve(allEnrolments and confidenceLevel) {
       case enrolments ~ userConfidence if userConfidence.level >= minimumConfidenceLevel =>
-        handleEnrolmentValidation(enrolments, affinityGroup, request, block)
+        handleEnrolmentValidation(enrolments, affinityGroup, sessionId, request)(block)
       case _ =>
-        logAndRedirectFromUrl(
-          s"$individualAuthLogString User has confidence level below 250, routing user to IV uplift.",
-          appConfig.incomeTaxSubmissionIvRedirect)
+        logger.info(s"$individualLogPrefix User has confidence level below 250, routing user to IV uplift.")
+        Future.successful(Redirect(appConfig.incomeTaxSubmissionIvRedirect))
     }
   }
 
-  private def handleEnrolmentValidation[A](enrolments: Enrolments, affinityGroup: AffinityGroup,
-                                           request: Request[A], block: AuthorisationRequest[A] => Future[Result])
-                                          (implicit hc: HeaderCarrier): Future[Result] = {
+  private def handleEnrolmentValidation[A](enrolments: Enrolments,
+                                           affinityGroup: AffinityGroup,
+                                           sessionId: String,
+                                           request: Request[A]
+                                          )(block: AuthorisationRequest[A] => Future[Result]): Future[Result] = {
     val optionalMtdItId: Option[String] = enrolmentGetIdentifierValue(Individual.key, Individual.value, enrolments)
     val optionalNino: Option[String] = enrolmentGetIdentifierValue(Nino.key, Nino.value, enrolments)
-
     (optionalMtdItId, optionalNino) match {
       case (Some(mtdItId), Some(nino)) =>
-        createAuthorisationRequest(mtdItId, None, nino, affinityGroup, request, block)
+        block(AuthorisationRequest(models.User(mtdItId, None, nino, affinityGroup.toString, sessionId), request))
       case (_, None) =>
-        logAndRedirectFromUrl(
-          s"$individualAuthLogString - No active session. Redirecting to sign-in URL.",
-          appConfig.signInUrl)
+        logger.info(s"$individualLogPrefix - No active session. Redirecting to sign-in URL.")
+        Future.successful(Redirect(appConfig.signInUrl))
```

Comment:
Just realised, this looks like a small bug. Because if there's no NINO the log message should be different, and also, I don't think throwing them back through sign-in would make sense. 

This isn't your problem though, and I don't think we should fix it as part of this. But it might make sense to create a GitHub issue and then a JIRA ticket to track this and think about how we'd actually want to handle this better.